tests/meta/compress/test_compress.py

```diff
@@ -0,0 +1,131 @@
+import pytest
+import os
+from pathlib import Path
+
+from cg.apps.scoutapi import ScoutAPI
+from cg.apps.hk import HousekeeperAPI
+from cg.apps.crunchy import CrunchyAPI
+from cg.meta.compress import CompressAPI
+
+
+def test_get_nlinks(compress_api, compress_test_dir):
+    """Test get_nlinks"""
+    # GIVEN a bam
+    file_path = compress_test_dir / "file"
+    file_path.touch()
+
+    # WHEN getting number of links to inode
+    nlinks = compress_api.get_nlinks(file_link=file_path)
+
+    # THEN number of links should be one
+    assert nlinks == 1
+
+    # WHEN adding a link
+    first_link = compress_test_dir / "link-1"
+    os.link(file_path, first_link)
+    nlinks = compress_api.get_nlinks(file_link=file_path)
+
+    # THEN number of links should be two
+    assert nlinks == 2
+
+    # WHEN adding yet another link
+    second_link = compress_test_dir / "link-2"
+    os.link(file_path, second_link)
+    nlinks = compress_api.get_nlinks(file_link=file_path)
+
+    # THEN number of links should be three
+    assert nlinks == 3
+
+
+def test_get_bam_files(compress_api, compress_scout_case, bam_files_hk_list, mocker):
+    """test get_bam_files method"""
+
+    # GIVEN a case id
+    case_id = "test_case"
+    mock_get_cases = mocker.patch.object(
+        ScoutAPI, "get_cases", return_value=[compress_scout_case]
+    )
+    mock_get_files = mocker.patch.object(
+        HousekeeperAPI, "get_files", return_value=bam_files_hk_list
+    )
+
+    # WHEN getting bam-files
+    bam_dict = compress_api.get_bam_files(case_id=case_id)
+
+    # THEN the bam-files for the samples will be in the dictionary
+    assert set(bam_dict.keys()) == set(["sample_1", "sample_2", "sample_3"])
+
+
+def test_compress_case_bams(compress_api, bam_dict, mocker):
+    """Test compress_case method"""
+    # GIVEN a compress api and a bam dictionary
+    mock_bam_to_cram = mocker.patch.object(CrunchyAPI, "bam_to_cram")
+
+    # WHEN compressing the case
+    compress_api.compress_case_bams(bam_dict=bam_dict, ntasks=1, mem=2)
+
+    # THEN all samples in bam-files will have been compressed
+    assert mock_bam_to_cram.call_count == len(bam_dict)
+
+
+def test_update_scout(compress_api, bam_dict, mock_compress_func, mocker):
+    """ Test update_hk method"""
+    mock_compress_func(bam_dict)
```

Comment:
I thought you could not call fixtures with arguments... Perhaps send in the bam_dict in the actual fixture?

Reply:
you can't, but that fixture returns a callable function 😄

Reply:
Haha a hack!